### PR TITLE
Migrated most service tests to use new expectBadge

### PR DESCRIPTION
**Pull request by PyvesB**
Follow up to #3097 and related to #2687. This pull request converts most service tests to use the new `expectBadge` helper, apart from:
* Wercker's tests which was completed as part of #3097.
* Bitrise's tests which was completed as part of #3117.
* the static-badge `Not a valid color` test. This checks for a `null` color, which is not currently supported by the `expectBadge` helper. I will take care of this is a subsequent pull request.
* two tests in `endpoint.tester.js`, which are checking for unusual fields such as `labelColor` or `logoWidth` in the JSON response. I was hoping that we could remove the `_shields_test` style by adding the `color` field to the JSON response when tackling #2686 and therefore have `expectBadge` only ever check for `label`, `message` and `color`, but these two special tests will prevent us from doing that. Thoughts?

Don't be too frightened by the size of the diff! Most changes are pretty straightforward and I've left a few explanations for the less obvious ones. Also, to make things easier for the reviewer I've only strictly modified the `*.tester.js` files in `services`. No changes to the service or helper code. 😉

### CODE REVIEW

File: services/nodeping/nodeping-status.tester.js [hunk 1]
```diff
@@ -3,13 +3,12 @@
 const Joi = require('joi')
 const t = (module.exports = require('../tester').createServiceTester())
 
-t.create('NodePing status - live').get(
-  '/jkiwn052-ntpp-4lbb-8d45-ihew6d9ucoei.json'
-)
-Joi.object().keys({
```
Comment (PyvesB) [inline services/nodeping/nodeping-status.tester.js]: This was incorrectly set up.

File: services/opencollective/opencollective-all.tester.js [hunk 2]
```diff
@@ -20,43 +19,35 @@ t.create('renders correctly')
...
-    })
-  )
+  .expectBadge({
+    label: 'backers and sponsors',
+    message: nonNegativeInteger,
+  })
 
 t.create('renders not found correctly')
-  .get('/nonexistent-collective.json?style=_shield_test')
+  .get('/nonexistent-collective.json?style=_shields_test')
```
Comment (PyvesB) [inline services/opencollective/opencollective-all.tester.js]: Several of the OpenCollective tests were missing the `s` in `_shields_test`, the new helper highlighted this mistake.

File: services/opencollective/opencollective-by-tier.tester.js [hunk 3]
```diff
@@ -1,11 +1,10 @@
 'use strict'
 
-const Joi = require('joi')
 const { nonNegativeInteger } = require('../validators')
 const t = (module.exports = require('../tester').createServiceTester())
 
 t.create('renders correctly')
-  .get('/shields/2988.json')
+  .get('/shields/2988.json?style=_shields_test')
```
Comment (PyvesB) [inline services/opencollective/opencollective-by-tier.tester.js]: Several of the OpenCollective tests were missing the `_shields_test` style entirely but were silently ignoring the color check, the new helper highlighted this mistake.

File: services/dependabot/dependabot.tester.js [hunk 4]
```diff
@@ -1,31 +1,26 @@
...
 const { isIntegerPercentage } = require('../test-validators')
 const t = (module.exports = require('../tester').createServiceTester())
 
 t.create('semver stability (valid)')
   .get('/bundler/puma.json')
-  .expectJSONTypes(
-    Joi.object().keys({
-      name: 'semver stability',
-      value: isIntegerPercentage,
-      link:
```
Comment (PyvesB) [inline services/dependabot/dependabot.tester.js]: This is not returned even with `_shields_test` specified.

**Comment by PyvesB** [comment]
Thanks for taking a look @chris48s. Things also look good on my side when I run the tests, most failures seem related to #3089. Merging!

**Comment by paulmelnikow** [comment]
Thanks for taking this on! Nice work!

**Pull request merged**